app/scripts/lib/operation-safener.ts

```diff
@@ -0,0 +1,137 @@
+import { type DebounceSettings, type DebouncedFunc, debounce } from 'lodash';
+import log from 'loglevel';
+
+export type { DebounceSettings } from 'lodash';
+
+// eslint-disable-next-line @typescript-eslint/no-explicit-any
+type AnyParams = any[];
+
+/**
+ * A type representing a generic operation that can be executed.
+ */
+export type Op = (...params: AnyParams) => unknown | Promise<unknown>;
+
+/**
+ * Options for the lock used in the OperationSafener.
+ */
+type Config<O extends Op> = {
+  /**
+   * The operation to be debounced and executed safely.
+   */
+  op: O;
+  /**
+   * The wait time in milliseconds for the debounce function. Defaults to 0ms.
+   */
+  wait?: number;
+  /**
+   * Additional options for the debounce function. See
+   * [lodash's debounce documentation](https://lodash.com/docs/4.17.15#debounce)
```

Comment:
In a future PR we can use the `maxWait` feature of lodash's `debounce`, which will force a write through every `maxWait`ms. Neat!